src/vstest.console/Publisher/NoOpMetricsPublisher.cs

```diff
@@ -17,7 +23,12 @@ public class NoOpMetricsPublisher : IMetricsPublisher
         /// <param name="metrics"></param>
         public void PublishMetrics(string eventName, IDictionary<string, object> metrics)
         {
-            // No Operation
+            // Log to Text File
+            var logEnabled = Environment.GetEnvironmentVariable("VSTEST_LOGTELEMETRY");
+            if (!string.IsNullOrEmpty(logEnabled) && logEnabled.Equals("1", StringComparison.Ordinal))
+            {
+                this.LogToFile(eventName, metrics, new FileHelper());
```

Comment:
Suggest to add a console output or trace the file name to allow correlation.

Reply:
Can we log the file in EqtTrace logs?